File: JS/LonelyInteger.test.js
/**
 * @version JAVASCRIPT ES6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { lonelyinteger } from './LonelyInteger.js'

TEST('Starting LonelyInteger test...', (t) => {
    //Test 1
    let want = 4
    let got = lonelyinteger( [1, 2, 3, 4, 3, 2, 1] )
    t.assert(want === got, "Expect: 4")
    //Test 2
    want = 2
    got = lonelyinteger( [0, 0, 1, 2, 1] )
    t.assert(want === got, "Expect: 2")
    //Test 3
    want = 2
    got = lonelyinteger( [1, 1, 2] )
    t.assert(want === got, "Expect: 2")
    //Test 4
    want = 2
    got = lonelyinteger( [1, 2, 3, 4, 3, 4, 1] )
    t.assert(want === got, "Expect: 2")

    t.end()
})

File: JAVA/LonelyIntegerTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import java.util.ArrayList;
import java.util.List;
import org.junit.Test;

public class LonelyIntegerTest {
    
    @Test
	public void testLonelyInteger() throws Exception {
		//Test 1
		int want = 4;
        List<Integer> ops = new ArrayList<Integer>();
        ops.add(1);
        ops.add(2);
        ops.add(3);
        ops.add(4);
        ops.add(3);
        ops.add(2);
        ops.add(1);
		int got = LonelyInteger.lonelyinteger( ops );
        assertEquals(want, got);
		//Test 2
		want = 2;
        ops = new ArrayList<>();
        ops.add(0);
        ops.add(0);
        ops.add(1);
        ops.add(2);
        ops.add(1);
		got = LonelyInteger.lonelyinteger( ops );
        assertEquals(want, got);
		//Test 3
		want = 2;
        ops = new ArrayList<>();
        ops.add(1);
        ops.add(1);
        ops.add(2);
		got = LonelyInteger.lonelyinteger( ops );
        assertEquals(want, got);
		//Test 4
		want = 2;
        ops = new ArrayList<>();
        ops.add(1);
        ops.add(2);
        ops.add(3);
        ops.add(4);
        ops.add(3);
        ops.add(4);
        ops.add(1);
		got = LonelyInteger.lonelyinteger( ops );
        assertEquals(want, got);
	}

}
